File: javascript/gigasecond/gigasecond.js
const secondsToMilliseconds = (seconds) => seconds * 1000; //Average: 1.89s
const oneGigasecond = Math.pow(10,9)
export const gigasecond = (data) => {
//	const secondsToMilliseconds = (seconds) => seconds * 1000; // Average: 2.09s
//	const oneGigasecond = Math.pow(10,9);
	return new Date(data.getTime() + secondsToMilliseconds(oneGigasecond)); //Date.getTime() return milliseconds, so we need to multiply the seconds by 1000
}

File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.util.Date;
import java.time.LocalDateTime;
import java.time.temporal.ChronoUnit;
class Gigasecond {
	final long oneGigasecond = (long) Math.pow(10,9);
	LocalDateTime datePlusGiga;

    Gigasecond(LocalDate moment) {
		this.datePlusGiga = moment.plus(oneGigasecond, ChronoUnit.SECONDS).atStartOfDay();	
	}

    Gigasecond(LocalDateTime moment) {
		this.datePlusGiga = moment.plusSeconds(oneGigasecond);
    }

    LocalDateTime getDateTime() {
		return this.datePlusGiga;
	}
}
